### ISSUE TITLE
Copy&paste does not assign new IDs

### ISSUE THREAD

**Issue by tomkrysl** [jmpessoa/lazandroidmodulewizard#25]
When copying eg. jEditText its ID (visible in lfm) does not change. This is a bug because then Anchor based sizing and positioning does not work as expected.

Quick fix: Make ID visible in object inspector to change it manually.
Better fix: Generate new ID when copied. Then traverse all controls and ensure they have unique ID.

**Comment by odisey1245** [comment]
lamwdesigner.pas:
procedure TAndroidWidgetMediator.OnPersistentAdded

Why IDs are generated in so ugly way? Why they cannot be generated as a sequence?

**Comment by jmpessoa** [comment]
Well,  now that property ID was made public  by @tomkrysl  [tk]   the "ugliness" has become visible. Can we [someone] improve this?

**Comment by tomkrysl** [comment]
I'll try to improve that now, it was only a quick&dirty fix. I think these should stay visible like TabOrder, to allow full control of them when sth. goes wrong.

**Comment by tomkrysl** [comment]
Should be now fixed. I made new TAndroidForm.AutoAssignIDs property. When AutoAssignIDs is set to True at design time then LAMW asks to reassign existing IDs instantly. This will make nice IDs. When answered no then LAMW makes nice IDs on next form open. When AutoAssignIDs is True then LAMW always reassigns these IDs on form open. Each visual control has nonzero ID now. I kept the IDs visible in object inspector, just to allow visual control about the IDs. IDs are still writable but when AutoAssignIDs = True all changes will be lost on next form open.
